src/DonationForms/V2/Endpoints/ListDonationForms.php

```diff
@@ -141,6 +142,7 @@ public function handleRequest(WP_REST_Request $request): WP_REST_Response
                 $item['permalink'] = get_permalink($item['id']);
                 $item['v3form'] = (bool)give_get_meta($item['id'], 'formBuilderSettings');
                 $item['status_raw'] = $forms[$i]->status->getValue();
+                $item['isDefaultCampaignForm'] = ($campaign = Campaign::findByFormId($item['id'])) && $item['id'] === $campaign->defaultForm()->id;
```

Comment:
This shouldn't be in the loop because it's creating n+1, and we want to avoid that. You can get the default form ID outside the loop and then just compare the item ID against the default form ID.

Reply:
Not sure if I understand it. How can I use the `Campaign::findByFormId($item['id'])` outside of the loop if the `$item` var is available only on the loop?

Reply:
All these forms are associated with one campaign, right? And we already have the campaign ID.

So instead of getting the campaign by form ID in a loop, we should get the campaign by campaign ID outside the loop. 

So instead of doing this:

```php            
foreach ($items as $i => &$item) {
      $item['isDefaultCampaignForm'] = ($campaign = Campaign::findByFormId($item['id'])) && $item['id'] === $campaign->defaultForm()->id;
} 
```

We should do something like this:

```php            
$defaultForm = Campaign::find($this->request->get_param('campaignId'))->defaultForm();

foreach ($items as $i => &$item) {
      $item['isDefaultCampaignForm'] = $item['id'] === defaultForm->id;
} 
```

The difference here is that we only make one request. What we have now is making requests for each form. So imagine someone has 100 forms, we'll be doing 100 unnecessary requests.

Reply:
Oh man, I totally ignored the `campaignId` parameter, now makes sense, thanks for noticing that! Resolved in e27a1c4d729ebe225064a9a5c69578743fcf1834

Reply:
Thanks, man!